keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""BERT task specific models and heads."""
+
+import copy
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_nlp.models.gpt2.gpt2_backbone import GPT2Backbone
+from keras_nlp.models.gpt2.gpt2_preprocessor import GPT2CausalLMPreprocessor
+from keras_nlp.models.gpt2.gpt2_presets import backbone_presets
+from keras_nlp.samplers.beam_sampler import BeamSampler
+from keras_nlp.samplers.greedy_sampler import GreedySampler
+from keras_nlp.samplers.top_k_sampler import TopKSampler
+from keras_nlp.samplers.top_p_sampler import TopPSampler
+from keras_nlp.utils.pipeline_model import PipelineModel
+from keras_nlp.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_nlp")
+class GPT2CausalLM(PipelineModel):
+    def __init__(self, backbone, preprocessor=None, **kwargs):
+
+        inputs = backbone.input
+        x = backbone(inputs)
+        x = tf.matmul(
+            x,
+            backbone.get_layer("token_embedding").embeddings,
+            transpose_b=True,
+        )
+        outputs = tf.keras.layers.Softmax()(x)
```

Comment:
I think we have to be consistent in our outputs across all our task models (or add a flag for this).

Right now our classifiers all return logits, so I think we need to return logits here too.